src/plugins/owner/ECDSAValidationPlugin.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+
+import {IPlugin} from "../../interfaces/IPlugin.sol";
+import {IValidation} from "../../interfaces/IValidation.sol";
+import {BasePlugin} from "../BasePlugin.sol";
+import {PluginManifest, PluginMetadata} from "../../interfaces/IPlugin.sol";
+
+contract ECDSAValidationPlugin is IValidation, BasePlugin {
+    using ECDSA for bytes32;
+    using MessageHashUtils for bytes32;
+
+    uint256 internal constant _SIG_VALIDATION_PASSED = 0;
+    uint256 internal constant _SIG_VALIDATION_FAILED = 1;
+
+    // bytes4(keccak256("isValidSignature(bytes32,bytes)"))
+    bytes4 internal constant _1271_MAGIC_VALUE = 0x1626ba7e;
+    bytes4 internal constant _1271_INVALID = 0xffffffff;
+
+    mapping(uint8 id => mapping(address account => address)) public owners;
+
+    error AlreadyInitialized();
+    error NotAuthorized();
+    error NotInitialized();
+
+    /// @inheritdoc IPlugin
+    function onInstall(bytes calldata data) external override {
+        uint8 id = uint8(bytes1(data[:1]));
+
+        if (owners[id][msg.sender] != address(0)) {
+            revert AlreadyInitialized();
+        }
+
+        address owner = abi.decode(data[1:], (address));
+        owners[id][msg.sender] = owner;
+    }
+
+    /// @inheritdoc IPlugin
+    function onUninstall(bytes calldata data) external override {
+        uint8 id = uint8(bytes1(data[:1]));
+
+        if (owners[id][msg.sender] == address(0)) {
+            revert NotInitialized();
+        }
+
+        delete owners[id][msg.sender];
+    }
+
+    /// @inheritdoc IValidation
+    function validateRuntime(uint8 functionId, address sender, uint256, bytes calldata, bytes calldata)
+        external
+        view
+        override
+    {
+        // TODO: not composable here, need to add a param to `validateRuntime` to pass in the account.
+        if (sender != owners[functionId][msg.sender]) {
+            revert NotAuthorized();
+        }
+        return;
```

Comment:
The risk of allow a param and have this function be called outside of account can be huge. However does it ensure the caller is sending the correct Account and sender since there is no UO or sig attached?
Example, a caller can sender over AccountA with SenderA for accountB.

Reply:
I think as long as this function is a view function (no state modifications), it's fine to parameterize the account instead of relying on `msg.sender`. If a validation function calls out to another validation function, it will be part of the internal logic of the outer validation function, and the account will still only trust the output of the function actually installted.

E.g. consider the scenario
```
account --calls--> validation function A --calls--> validation function B
```

In this setup, validation function A and B can both take in a parameterized sender. The account will send its own address, and validation function A _should_ send the same account address. But even if it doesn't, there's no security vulnerability or escalation - the account trusts A, but does not trust B to be a validation function.

Reply:
Right. I think in RI, we are covered. Could there be out of context usage of `validateRuntime`?
Maybe a word of caution is enough.

Reply:
I think we block it from being installed, because it's a known selector.

Other than that, I think we should be good? If someone calls it directly with `execute`/`executeBatch`, it shouldn't do anything.